### PR TITLE
Use `exec` in the CLI shell scripts to prevent new process creation

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description

OSD 2.6 in prior used `exec` in the CLI scripts to invoke `node`; this was removed when `use_node` was introduced. This change reintroduces `exec` which will prevent new processes from being created.

`exec` executes a command that completely replaces the current process where the current shell process is destroyed. When you exec a command, it replaces bash entirely — no new process is forked, no new PID is created, and all memory controlled by bash is destroyed and overwritten ([ref](https://www.computerhope.com/unix/bash/exec.htm)). This is useful for when the `bin` file calls `use_node` who then calls `node`; instead of having 3 PIDs to check for health, 2 of which will be doing nothing, `exec` allows `use_node` to replace the `bin` script and then `node` to replace the `use_node`, leaving us with just one PID.

### Check List

- [X] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [X] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by seraphjiang — APPROVED** [review]

**Comment by seraphjiang** [comment]
@himsgupta1122 @ZilongX @zhyuanqi FYI

**Comment by seraphjiang** [comment]
Thanks @AMoo-Miki for the change to unblock downsteam usage. Maintainers, could we quick review and approve it. it is critical, could we backport and add to 2.7 release

cc: @seanneumann  @bbarani  @zelinh

**Comment by joshuarrrr** [comment]
> This change reintroduces exec which will prevent new processes from being created.

There's no information provided here about why this is necessary or beneficial. There should be a linked issue that describes the problem and or regression.

**Review by manasvinibs — DISMISSED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Review by ananzh — APPROVED** [review]
I think this is from the discussion that we had in node 18 PR that `exec` could allow the new process keeps the same process ID as the original shell process. This can make process management easier, as you don't need to keep track of a new process ID. If I am wrong, correct me.

**Review by manasvinibs — APPROVED** [review]

**Comment by AMoo-Miki** [comment]
Will merge this despite the Cypress tests failing as they cannot be related.

**Pull request merged**
